src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -746,12 +748,19 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
                     return;
                 }
 
-                // do not send an invocation request if cancellation has been requested
                 if (context.CancellationToken.IsCancellationRequested)
                 {
-                    _workerChannelLogger.LogWarning("Cancellation has been requested. The invocation request with id '{invocationId}' is canceled and will not be sent to the worker.", invocationId);
-                    context.ResultSource.TrySetCanceled();
-                    return;
+                    _workerChannelLogger.LogInformation("Cancellation was requested prior to the invocation request ('{invocationId}') being sent to the worker. " +
+                                                        "For more information see https://aka.ms/AAmnbwj", invocationId);
+
+                    // If a worker is not configured to handle already cancelled invocations, do not
+                    // send an invocation request to the worker.
+                    if (!_isHandlesCanceledInvocationRequestsCapabilityEnabled)
```

Comment:
I agree with @jviau on this one -- it feels wrong to send a known-cancelled function invocation to the worker. It was cancelled for a reason (host is shutting down, client disconnected, etc)... and seems more confusing to let a worker start processing that invocation when it may never complete.